tests/components/blebox/test_init.py

```diff
@@ -0,0 +1,54 @@
+"""BleBox devices setup tests."""
+
+import logging
+
+import blebox_uniapi
+
+from homeassistant.components.blebox.const import DOMAIN
+
+from .conftest import mock_config, patch_product_identify
+
+
+async def test_setup_failure(hass, caplog):
+    """Test that setup failure is handled and logged."""
+
+    patch_product_identify(None, side_effect=blebox_uniapi.error.ClientError)
+
+    entry = mock_config()
+    entry.add_to_hass(hass)
+
+    caplog.set_level(logging.ERROR)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert "Identify failed at 172.100.123.4:80 ()" in caplog.text
```

Comment:
We can also assert that the `entry.state` is `ENTRY_STATE_SETUP_RETRY` in this case.